mppe/crawler.go

```diff
@@ -0,0 +1,408 @@
+			},
+		},
+		"servidoresInativos": {
+			category:     "proventos-de-todos-os-servidores-inativos",
+			pathResolver: inactiveEmployeesPathResolver,
+			yearCodes: map[int]int{
+				2011: 297,
+				2012: 298,
+				2013: 299,
+				2014: 300,
+				2015: 301,
+				2016: 302,
+				2017: 345,
+				2018: 408,
+				2019: 448,
+				2020: 507,
+			},
+		},
+		"pensionistas": {
+			category:     "valores-percebidos-por-todos-os-pensionistas",
+			pathResolver: pensionersPathResolver,
+			yearCodes: map[int]int{
+				2011: 303,
+				2012: 304,
+				2013: 305,
+				2014: 306,
+				2015: 307,
+				2016: 308,
+				2017: 346,
+				2018: 409,
+				2019: 449,
+				2020: 508,
+			},
+		},
+		"colaboradores": {
+			category:     "valores-percebidos-por-todos-os-colaboradores",
+			pathResolver: partnersPathResolver,
+			yearCodes: map[int]int{
+				2016: 365,
+				2017: 366,
+				2018: 410,
+				2019: 450,
+				2020: 496,
+			},
+		},
+		"exerciciosAnteriores": {
+			category:     "verbas-referentes-a-exercicios-anteriores",
+			pathResolver: previousYearsPatternToSearch,
+			yearCodes: map[int]int{
+				2016: 348,
+				2017: 349,
+				2018: 411,
+				2019: 461,
+				2020: 509,
+			},
+		},
+		"indenizacoesEOutrosPagamentos": {
+			category:     "verbas-indenizatorias-e-outras-remuneracoes-temporarias",
+			pathResolver: indemnityAndOtherPaymentsPatternToSearch,
+			yearCodes: map[int]int{
+				2018: 415,
+				2019: 451,
+				2020: 510,
+			},
+		},
+	}
+)
+
+// Crawl download all files related to the MPPE salaries and return their local paths
+func Crawl(outputPath string, month, year int) ([]string, error) {
+	paths := make([]string, 8)
+
+	pathChannel := make(chan string, 8)
+
+	errChannel := make(chan error, 8)
+
+	var wg sync.WaitGroup
+
+	for _, member := range members {
+		wg.Add(1)
+		go func(member fileStructure, month, year int) {
+			defer wg.Done()
+			link := getURLForYear(year, member.category, member.yearCodes)
+			resp, err := http.Get(link)
+			if err != nil {
+				errChannel <- fmt.Errorf("error getting downloading main html file :%q", err)
+			}
+			defer resp.Body.Close()
+			b, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				errChannel <- fmt.Errorf("error reading response body: %q", err)
+			}
+			htmlAsString := string(b)
+			pattern := member.pathResolver(month, year)
+			fileCode, err := findFileIdentifier(htmlAsString, pattern)
+			if err != nil {
+				errChannel <- err
+			}
+			urlToDownload := getURLToDownloadSheet(year, member.category, fileCode, member.yearCodes)
+			filePath := getFileName(member.category, outputPath, month, year)
+			desiredFile, err := os.Create(filePath)
+			if err != nil {
+				errChannel <- fmt.Errorf("error creating sheet file:%q", err)
+			}
+			defer desiredFile.Close()
+			err = donwloadFile(urlToDownload, desiredFile)
+			if err != nil {
+				errChannel <- fmt.Errorf("error downloading main file: %s %q", filePath, err)
+			}
+			pathChannel <- filePath
+		}(member, month, year)
+	}
+	wg.Wait()
```

Comment:
coloca o wait e os closes em uma go-routine. Dessa forma não precisará processar todos os arquivos para começar (já vai direto para o for).

```
go func(){
  wg.Wait()
  close(errChannel)
  close(pathChannel)
}()
```